lib/LaTeXML/Package/TeX.pool.ltxml

```diff
@@ -213,13 +213,14 @@ DefParameterType('XUntil', sub {
 # This is sorta like readbalanced, but expands as it goes.
 # This appears to be needed by certain primitives (eg. \noalign ?)
 # and maybe what we should be using for some Digested ??
+# and maybe we have to remove any "dont_expand" tokens here if we are to digest twice?
 DefParameterType('Expanded', sub {
     my ($gullet) = @_;
     my $token    = $gullet->readXToken(0);
     my @tokens   = ();
     if ($token->getCatcode == CC_BEGIN) {
       my $level = 1;
-      while ($token = $gullet->readXToken(0)) {
+      while ($token = $gullet->readXToken(0)->without_dont_expand) {
```

Comment:
This is one of the head-scratchers. Clearly we must not remove any flags in `readXToken` since that token may be used in all kinds of scenarios -- very importantly in a `ExpandedIfToken` parameter type, which then does very precise comparisons on the flagged tokens for the conditional's sake.

However, uses of `\expanded{\expanded{...}}` do indeed expand *twice*, so the flag needs to be removed in whichever parameter type `\expanded` is using (or in some of the gullet calls of said parameter type). This refactor seems to be getting a little closer to the "spirit" of these two particular affairs.